python/pyspark/sql/tests.py

```diff
@@ -476,6 +476,19 @@ def test_udf_defers_judf_initalization(self):
             "judf should be initialized after UDF has been called."
         )
 
+    @unittest.skipIf(sys.version_info < (3, 3), "Unittest < 3.3 doesn't support mocking")
```

Comment:
@holdenk I believe that for a full test `udf` would have to create `SparkContext`. But mock is cheap.